inference-engine/include/ie_compound_blob.h

```diff
@@ -349,4 +265,46 @@ class INFERENCE_ENGINE_API_CLASS(I420Blob) : public CompoundBlob {
 
     Blob::Ptr createROI(const ROI& roi) const override;
 };
+
+/**
+ * @brief This class represents a blob that contains other blobs - one per batch
+ */
+class INFERENCE_ENGINE_API_CLASS(BatchedBlob) : public CompoundBlob {
+ public:
+    /**
+     * @brief A smart pointer to the BatchedBlob object
+     */
+    using Ptr = std::shared_ptr<BatchedBlob>;
+
+    /**
+     * @brief A smart pointer to the const BatchedBlob object
+     */
+    using CPtr = std::shared_ptr<const BatchedBlob>;
+
+    /**
+     * @brief Constructs a batched blob from a vector of blobs
+     * @details All passed blobs should meet following requirements:
+     * - all blobs have equal tensor descriptors,
+     * - blobs layouts should be one of: NCHW, NHWC, NCDHW, NDHWC, NC, CN, C, CHW
+     * - batch dimensions should be equal to 1 or not defined (C, CHW).
+     * Resulting blob's tensor descriptor is constructed using tensor descriptors
+     * of passed blobs by setting batch dimension to blobs.size()
+     *
+     * @param blobs A vector of blobs that is copied to this object
+     */
+    explicit BatchedBlob(const std::vector<Blob::Ptr>& blobs);
+
+    /**
+     * @brief Constructs a batched blob from a vector of blobs
+     * @details All passed blobs should meet following requirements:
+     * - all blobs have equal tensor descriptors,
+     * - blobs layouts should be one of: NCHW, NHWC, NCDHW, NDHWC, NC, CN, C, CHW
+     * - batch dimensions should be equal to 1 or not defined (C, CHW).
+     * Resulting blob's tensor descriptor is constructed using tensor descriptors
+     * of passed blobs by setting batch dimension to blobs.size()
+     *
+     * @param blobs A vector of blobs that is moved to this object
+     */
+    explicit BatchedBlob(std::vector<Blob::Ptr>&& blobs);
```

Comment:
Nitpick: I'd use *sink* argument here - pass by value + `std::move`. It'll allow to have one ctor instead of 2 with same performance

Reply:
Thank you.
Unfortunately, copy-ellision is not guaranteed in here until C++17

Reply:
Which exactly copy you expect to be elided? I mean if we have argument by value, so in case of l-value argument call we have exactly 1 copy (the same for const reference version).